### PR TITLE
Twitch.tv provider

### PR DESCRIPTION
**Pull request by themasch**
I added this very basic oauth2 provider for twitch.tv. 
Docs for the twitch.tv api can be found here: https://github.com/justintv/Twitch-API/blob/master/authentication.md

I might improve this provider in the future (like using `/user/:username` instead of `/user` to get the profile if user_read isn't set). But it should work good as it is now.

### CODE REVIEW

File: lib/providers/twitch.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+var Hoek = require('hoek');
```
Comment (geek) [inline lib/providers/twitch.js]: Please add to the top:
```
// Load modules
```

File: lib/providers/twitch.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+var url = 'https://api.twitch.tv/kraken/';
```
Comment (geek) [inline lib/providers/twitch.js]: Please use the url string value in the return object below instead of a variable.

File: lib/providers/twitch.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+            // twitch requires 'OAuth' instead of 'Bearer'
+            var twOptions = {
+                headers: {
+                    Authorization: 'OAuth ' + params.access_token
+                }
+            };
+            Hoek.merge(this.provider, twOptions);
+
+            get(url + 'user', {}, function(body) {
```
Comment (geek) [inline lib/providers/twitch.js]: `function (body) {`

File: lib/providers/twitch.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+                credentials.profile = body;
```
Comment (geek) [inline lib/providers/twitch.js]: newline after function declarations

**Comment by geek** [comment]
Looks good.  Please make the minor changes and remerge and its good to go!

**Comment by geek** [comment]
@themasch do you think you will have time to get this PR updated?

**Comment by themasch** [comment]
Meh, the merge broke stuff...

**Comment by geek** [comment]
@themasch ya, you may be better off making a new PR after pulling the latest into a new branch and applying your changes.

**Comment by themasch** [comment]
okay, i'll do that then. I close this one here, not needed anymore.

**Pull request closed without merging**